src/test/java/de/tum/in/www1/artemis/AccountResourceWithGitLabIntegrationTest.java

```diff
@@ -0,0 +1,80 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+
+import de.tum.in.www1.artemis.connector.gitlab.GitlabRequestMockProvider;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.ModelFactory;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+import de.tum.in.www1.artemis.web.rest.vm.ManagedUserVM;
+
+public class AccountResourceWithGitLabIntegrationTest extends AbstractSpringIntegrationJenkinsGitlabTest {
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    UserRepository userRepo;
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    private GitlabRequestMockProvider gitlabRequestMockProvider;
+
+    @AfterEach
+    public void tearDown() {
+        database.resetDatabase();
+        gitlabRequestMockProvider.reset();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        gitlabRequestMockProvider.enableMockingOfRequests();
+    }
```

Comment:
```suggestion
    @BeforeEach
    public void setUp() {
        gitlabRequestMockProvider.enableMockingOfRequests();
    }

    @AfterEach
    public void tearDown() {
        database.resetDatabase();
        gitlabRequestMockProvider.reset();
    }
```
I am not sure here, this might be highly opinionated, but I think swapping `@BeforeEach` and `@AfterEach` here would make the code quicker to read.